### PR TITLE
Add fb and adwords tracking for akismet purchases

### PR DESCRIPTION
**Pull request by CodeyGuyDylan**
This diff adds FB and Adwords tracking for akismet purchases in the same way Jetpack purchases are tracked

P2: pbNhbs-5df-p2

## Proposed Changes

* Add new Akismet constants to the `calypso-products` package
* Identify Akismet products in cart splitting function in order to conditionally send tracking events
* Add tracking events for akismet products

## Testing Instructions

For reference when testing, these are the product slugs for Akismet products
* ak_plus_yearly_1
* ak_plus_monthly_1
* ak_plus_yearly_2
* ak_plus_monthly_2
* ak_plus_yearly_3
* ak_plus_monthly_3
* ak_plus_yearly_4
* ak_plus_monthly_4
* ak_ent_yearly_1
* ak_ent_monthly_1
* ak_ep350k_yearly_1
* ak_ep350k_monthly_1
* ak_ep2m_yearly_1
* ak_ep2m_monthly_1
* ak_epgt2m_yearly_1
* ak_epgt2m_monthly_1

1. Checkout this branch
2. Run `yarn start`
3. Point `public-api.wordpress.com` to your sandbox and enable store sandbox
4. Make sure you have a test site that you can attach the akismet license to in order to complete the checkout ( I used a personal wordpress site WITHOUT JETPACK, for some reason I could not get this purchase to go through on an atomic site with Jetpack )
5. Enable ad-tracking in your config (can be found here: fbhepr%2Skers%2Spnylcfb%2Spbasvt%2Sqrirybczrag.wfba%3Se%3Qr497735s%2329-og)
6. Go to `http://calypso.localhost:3000/checkout/{product_slug}` (for product_slug, see reference above)
7. You will have to pick a site to attach the license to. Once you are redirected to the cart, add `?flags=ad-tracking,cookie-banner` to the end of the URL and refresh
8. Ensure that you allow for advertising trackers
Run
`document.cookie = 'sensitive_pixel_options=%7B%22ok%22%3Atrue%2C%22buckets%22%3A%7B%22essential%22%3Atrue%2C%22analytics%22%3Atrue%2C%22advertising%22%3Atrue%7D%7D'`
in the console
9. Run `localStorage.setItem( 'debug', 'calypso:analytics:*' );` in the console
10. Go through with the purchase
11. Confirm you see an Adwords event similar to this ( I have not confirmed the full tracking ID for this, this is in progress )
image
12. Confirm you see a Facebook event similar to this ( ensure the 2nd item in the object is tracking ID: 485349158311379 )
image
13. Ensure the same events for Jetpack are NOT recorded in addition to these

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/lib/analytics/ad-tracking/record-order.js [hunk 1]
```diff
@@ -378,7 +400,7 @@ function recordOrderInGoogleAds( cart, orderId, wpcomJetpackCartInfo ) {
 	}
 
 	// MCC-level event.
-	// @TODO Separate WPCOM from Jetpack events.
```
Comment (CodeyGuyDylan) [inline client/lib/analytics/ad-tracking/record-order.js:381]: Removed this comment as this has been done

File: client/lib/analytics/utils/split-wpcom-jetpack-cart-info.ts [hunk 2]
```diff
@@ -1,40 +1,54 @@
...
 	containsJetpackProducts: boolean;
+	containsWpcomProducts: boolean;
+	akismetCost: number;
 	jetpackCost: number;
 	wpcomCost: number;
+	akismetCostUSD: number;
 	jetpackCostUSD: number;
 	wpcomCostUSD: number;
 	totalCostUSD: number;
 };
```
Comment (CodeyGuyDylan) [inline client/lib/analytics/utils/split-wpcom-jetpack-cart-info.ts:19]: Order here may look weird, I just reordered to have the categories listed in this order for all types of data
1. Akismet
2. Jetpack
3. WPCOM

File: packages/calypso-products/src/constants/akismet.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+export const PRODUCT_AKISMET_PLUS_40K_YEARLY = 'ak_plus_yearly_4';
+export const PRODUCT_AKISMET_PLUS_40K_MONTHLY = 'ak_plus_monthly_4';
+export const PRODUCT_AKISMET_ENTERPRISE_YEARLY = 'ak_ent_yearly_1';
+export const PRODUCT_AKISMET_ENTERPRISE_MONTHLY = 'ak_ent_monthly_1';
+export const PRODUCT_AKISMET_ENTERPRISE_350K_YEARLY = 'ak_ep350k_yearly_1';
+export const PRODUCT_AKISMET_ENTERPRISE_350K_MONTHLY = 'ak_ep350k_monthly_1';
+export const PRODUCT_AKISMET_ENTERPRISE_2M_YEARLY = 'ak_ep2m_yearly_1';
+export const PRODUCT_AKISMET_ENTERPRISE_2M_MONTHLY = 'ak_ep2m_monthly_1';
+export const PRODUCT_AKISMET_ENTERPRISE_GT2M_YEARLY = 'ak_epgt2m_yearly_1';
+export const PRODUCT_AKISMET_ENTERPRISE_GT2M_MONTHLY = 'ak_epgt2m_monthly_1';
```
Comment (jboland88) [inline packages/calypso-products/src/constants/akismet.ts:19]: Should we add constants here for the Personal plans as well?

File: client/lib/analytics/utils/split-wpcom-jetpack-cart-info.ts [hunk 4]
```diff
@@ -1,40 +1,54 @@
...
+export function splitCartProducts( cart: ResponseCart ): WpcomJetpackCartInfo {
 	const jetpackCost = cart.products
 		.map( ( product ) =>
 			isJetpackPlan( product ) || isJetpackProduct( product ) ? product.cost : 0
 		)
 		.reduce( ( accumulator, cost ) => accumulator + cost, 0 );
+	const akismetCost = cart.products
+		.map( ( product ) => ( isAkismetProduct( product ) ? product.cost : 0 ) )
+		.reduce( ( accumulator, cost ) => accumulator + cost, 0 );
 	const wpcomCost = cart.total_cost_integer / 100 - jetpackCost;
```
Comment (jboland88) [inline client/lib/analytics/utils/split-wpcom-jetpack-cart-info.ts]: Does `wpcomCost` now also need to subtract `akismetCost`?

**Review by jboland88 — CHANGES_REQUESTED** [review]
Overall, this tested well and is looking good!
I noticed in testing that I also got a `Record WPCom Purchase` event, but I think this is expected, since that looks to fire for any purchase?
Screen Shot 2023-02-17 at 9 47 10 AM

I did find one small item and had one question.

**Comment by CodeyGuyDylan** [comment]
> I noticed in testing that I also got a Record WPCom Purchase event, but I think this is expected, since that looks to fire for any purchase?

That is what was previously in the logic, so I also somewhat assumed that it was expected behavior

File: client/lib/analytics/utils/split-wpcom-jetpack-cart-info.ts (hunk 4 above)
Comment (CodeyGuyDylan) [reply client/lib/analytics/utils/split-wpcom-jetpack-cart-info.ts]: Good catch, I didn't notice that

**Review by jboland88 — APPROVED** [review]
LGTM! Thanks for the updates!

**Comment by CodeyGuyDylan** [comment]
@gmovr I have updated the tracking ID, should be ready for you to take a look at 😄

**Comment by CodeyGuyDylan** [comment]
Updated the GTAG event parameter names to match what they currently are on the old website
Reference: https://docs.google.com/document/d/1VEnuC48taIlu8I3WfmN0SJ-STU1rqzhPYhECKLB8N-E/edit

File: packages/calypso-products/src/constants/akismet.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
+export const AKISMET_FREE = 'ak_free_yearly';
+export const AKISMET_PERSONAL_MONTHLY = 'ak_personal_monthly';
+export const AKISMET_PERSONAL_YEARLY = 'ak_personal_yearly';
```
Comment (robertsreberski) [inline packages/calypso-products/src/constants/akismet.ts]: Shouldn't we define these constants with `PRODUCT_` prefix as well?

**Comment by robertsreberski** [comment]
The code looks good to me! I left one small code change that doesn't impact the logic.

**Review by robertsreberski — APPROVED** [review]
The code looks good to me! I left one small code change that doesn't impact the logic.

**Review by robertsreberski — COMMENTED** [review]
I think we can also consider firing `page_view` events for calypso checkout pages related to Akismet, if possible. That will give us even more insight into user path.

File: packages/calypso-products/src/constants/akismet.ts (hunk 5 above)
Comment (CodeyGuyDylan) [reply packages/calypso-products/src/constants/akismet.ts]: Nice catch, yes they should be

**Pull request merged**